src/main/java/com/openshift/jenkins/plugins/pipeline/OpenShiftExec.java

```diff
@@ -0,0 +1,130 @@
+package com.openshift.jenkins.plugins.pipeline;
+
+import com.openshift.jenkins.plugins.pipeline.model.GlobalConfig;
+import com.openshift.jenkins.plugins.pipeline.model.IOpenShiftApiObjHandler;
+import com.openshift.jenkins.plugins.pipeline.model.IOpenShiftExec;
+import hudson.Extension;
+import hudson.model.AbstractProject;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.Builder;
+import hudson.util.FormValidation;
+import net.sf.json.JSONObject;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.StaplerRequest;
+
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+public class OpenShiftExec extends OpenShiftBaseStep implements IOpenShiftExec, IOpenShiftApiObjHandler {
+
+    protected final String pod;
+    protected final String container;
+    protected final String command;
+    protected final List<Argument> arguments;
+    protected final String waitTime;
+
+    public String getPod() {
+        return pod;
+    }
+
+    public String getContainer() {
+        return container;
+    }
+
+    public String getCommand() {
+        return command;
+    }
+
+    public List<Argument> getArguments() {
+        return arguments;
+    }
+
+    public String getWaitTime() {
+        return waitTime;
+    }
+
+    @DataBoundConstructor
+    public OpenShiftExec(String apiURL, String namespace, String authToken, String verbose, String pod, String container, String command, List<Argument>arguments, String waitTime) {
+        super( apiURL, namespace, authToken, verbose );
+        this.pod = pod.trim();
+        this.container = container.trim();
+        this.command = command;
+        this.arguments = arguments;
+        this.waitTime = waitTime.trim();
+    }
+
+    // TODO: Copied from other class boilerplate, but I don't think this works. We are passing a user specified timeout string as a map key.
+    public String getWaitTime(Map<String,String> overrides) {
```

Comment:
@oatmealraisin 's PR is reworking this, creating a more common layer.  Depending on how merges first, some small refactor will be needed by the other PR to include this with the wait time stuff.

The map as returned by getOverride for wait time deals with the user using a Jenkins property in the config field, where at the top of the job, or at the global config panel, that property is assigned the number desired.

I'm pretty sure it worked originally when first introduced, but if you think you've found a bug, we can discuss.